### DISCUSSION TITLE
Can't override template of a slider using a control theme?

### DISCUSSION THREAD

**Discussion by Omsad** [AvaloniaUI/Avalonia#13714] [Q&A] [answered]
## Describe the bug
I'm attempting to override the template of a slider using a control theme.

## To Reproduce
See [Example.zip](https://github.com/AvaloniaUI/Avalonia/files/13452018/Example.zip)

## Expected behavior
The slider should use the new template based upon the specified control theme.

## Environment
- OS: Windows
- Avalonia-Version: 11.0.5

## Additional context
I was attempting to change the margin of the slider so the ticks marks would align correctly, see: https://github.com/amwx/FluentAvalonia/issues/496 , but the only way I could get my template to work was by setting it explicitly, e.g.

```
if (slider.FindResource(typeof(Slider)) is ControlTheme controlTheme)
{
    slider.Theme = controlTheme;
    slider.Template = controlTheme.Setters
                                  .OfType<Setter>()
                                  .FirstOrDefault(match => match.Property?.Name == nameof(Template))
                                  ?.Value as IControlTemplate;
}
```

Without this the slider template doesn't change.
This happens even when not using FluentAvaloniaUI.

**Comment by MrJul** [comment, marked as answer]
You're basing your `ControlTheme` on the existing one, via `BasedOn="{StaticResource {x:Type Slider}}"`, so existing styles from there apply.

A simple change is to wrap your setter inside `<Style Selector="^:horizontal">` to override that existing style:
```xaml
<ControlTheme [...]>
  <Style Selector="^:horizontal">
    <Setter Property="Template">
      [...]
    </Setter>
  </Style>
</ControlTheme>
```

That said, you can simply forget about the `ControlTheme` if you're not redefining everything and use a standard `Style` instead:
```xaml
<Style Selector="Slider:horizontal">
```

I'm converting this issue to a discussion.

**Reply by Omsad** [reply]
Thanks, I had missed that the base control theme defines the template within <Style Selector="^:horizontal"> , after doing the same its working.
